### ISSUE TITLE
Add tests for the case when nodes are other than int, str

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#474]
In the case of DBNs, the nodes are tuples. Need to add tests for such nodes everywhere.

**Comment by kshitij10496** [comment]
@ankurankan I am interested in working on this.
Doesnt the `test_DynamicBayesianNetwork`[Link](https://github.com/pgmpy/pgmpy/blob/dev/pgmpy/tests/test_models/test_DynamicBayesianNetwork.py) have all the necessary tests ?
If not, what all things do I need to add ?

**Comment by ankurankan** [comment]
@kshitij10496 In pgmpy, ideally the nodes of any model can be any hashable python object i.e str, int, float, tuple etc. But most of the tests for models have been written using names of type str. So, we need to add more tests using int or tuples as node names.

**Comment by kshitij10496** [comment]
@ankurankan Okay.
But would it make a difference if we write test cases using any object as long as its hashable ?

**Comment by ankurankan** [comment]
@kshitij10496 Yes. Because earlier we used to only support objects of type str as node names. So, we want to make sure that we haven't missed changing something somewhere and is still not on the old behavior.
